kv.go

```diff
@@ -138,6 +139,45 @@ func (k *KVService) CompareAndDelete(key string, oldValue interface{}) (bool, er
 	return k.Set(key, nil, SetAtomic(oldValue))
 }
 
+// SetAtomicWithRetries will set a key-value pair atomically using compare and set semantics:
+// it will read key's value (to get oldValue), perform valueFunc (to get newValue),
+// and compare and set (comparing oldValue and setting newValue).
+//
+// Parameters:
+// `key` is the key to get and set.
+// `valueFunc` is a user-provided function that will take the old value and return the new value
+//    or an error.
+// `numRetries is the number of times the function will attempt to compare and set before failing.
+//
+// Returns err if the key could not be retrieved (DB error), valueFunc returned an error,
+//   if the key could not be set (DB error), or if the key could not be set (after retries)
+// Returns nil if the value was set
+//
+// Minimum server version: 5.18
+func (k *KVService) SetAtomicWithRetries(key string, valueFunc func(oldValue interface{}) (newValue interface{}, err error), numRetries int) error {
```

Comment:
Does numRetries need to be an option? It doesn't seem situational. Maybe we should just be opinionated about it.

Reply:
Yah, that makes sense. Will do.